src/shared_modules/rsync/src/rsyncImplementation.cpp

```diff
@@ -0,0 +1,321 @@
+                }
+            }
+            
+            ++index;
+        }
+    };
+
+    auto rowFilter { querySelect.at("row_filter").get_ref<const std::string&>() } ;
+    Utils::replaceFirst(rowFilter, "?", begin);
+    Utils::replaceFirst(rowFilter, "?", end);
+
+    auto& queryParam { selectData["query"] };
+    queryParam["row_filter"] = rowFilter;
+    queryParam["column_list"] = querySelect.at("column_list");
+    queryParam["distinct_opt"] = querySelect.at("distinct_opt");
+    queryParam["order_by_opt"] = querySelect.at("order_by_opt");
+
+    const std::unique_ptr<cJSON, CJsonDeleter> spJson{ cJSON_Parse(selectData.dump().c_str()) };
+    spDBSyncWrapper->select(spJson.get(), { callbackDBSync, &calcChecksum });
+}
+
+nlohmann::json RSyncImplementation::getRowData(const std::shared_ptr<DBSyncWrapper>& spDBSyncWrapper, 
+                                               const nlohmann::json& jsonSyncConfiguration,
+                                               const std::string& index)
+{
+    nlohmann::json rowData;
+    std::function<void(const nlohmann::json&)> getRowData
+    {
+        [&rowData] (const nlohmann::json& resultJSON)
+        {
+            rowData = resultJSON;       
+        }
+    };
+
+    nlohmann::json selectData;
+    selectData["table"] = jsonSyncConfiguration.at("table");
+    auto& queryParam { selectData["query"] };
+    const auto& querySelect { jsonSyncConfiguration.at("row_data_query_json") };
+
+    auto rowFilter { querySelect.at("row_filter").get_ref<const std::string&>() } ;
+    Utils::replaceFirst(rowFilter, "?", index);
+    
+    queryParam["row_filter"] = rowFilter;
+    queryParam["column_list"] = querySelect.at("column_list");
+    queryParam["distinct_opt"] = querySelect.at("distinct_opt");
+    queryParam["order_by_opt"] = querySelect.at("order_by_opt");
+
+    const std::unique_ptr<cJSON, CJsonDeleter> spJson{ cJSON_Parse(selectData.dump().c_str()) };
+
+    spDBSyncWrapper->select(spJson.get(), { callbackDBSync, &getRowData });
+    return rowData;
+}
+
+void RSyncImplementation::sendAllData(const std::shared_ptr<DBSyncWrapper>& spDBSyncWrapper, 
+                                      const nlohmann::json& jsonSyncConfiguration,
+                                      const ResultCallback callbackWrapper)
+{
+    const auto& messageCreator { FactoryMessageCreator<nlohmann::json, MessageType::ROW_DATA>::create() };
+    std::function<void(const nlohmann::json&)> sendRowData
+    {
+        [&callbackWrapper, &messageCreator, &jsonSyncConfiguration] (const nlohmann::json& resultJSON)
+        {
+            messageCreator->send(callbackWrapper, jsonSyncConfiguration, resultJSON.at(0));  
+        }
+    };
+    nlohmann::json selectData;
+    selectData["table"] = jsonSyncConfiguration.at("table");
+    const auto& querySelect { jsonSyncConfiguration.at("no_data_query_json") };
+
+    auto& queryParam { selectData["query"] };
```

Comment:
selectData.at("query")

Reply:
.at return a const and is only for read, I need append into queryParam in the future.